pkg/operator/ceph/cluster/osd/provision_spec.go

```diff
@@ -155,12 +155,11 @@ func (c *Cluster) provisionPodTemplateSpec(osdProps osdProperties, restart v1.Re
 	}
 
 	p := cephv1.GetOSDPlacement(c.spec.Placement)
-	if !osdProps.onPVC() {
-		p.ApplyToPodSpec(&podSpec)
-	} else {
-		osdProps.getPreparePlacement().ApplyToPodSpec(&podSpec)
-		p.ApplyToPodSpec(&podSpec)
+	if osdProps.onPVC() {
+		osdProps.getPreparePlacement().ApplyToPodSpec(&podSpec, false)
 	}
+	p.ApplyToPodSpec(&podSpec, false)
```

Comment:
How about adding a comment about why we don't want to merge the osd placement, since we don't want the placement for osds on pvcs and non-pvcs to overlap.

Reply:
Added.